draft-jholland-quic-multicast.md

```diff
@@ -156,95 +156,124 @@ A client has the option of refusal and the power to impose upper bound maxima on
+"able to join" means:
+- Both MC_KEY and MC_ANNOUNCE have been received
+- Result will be within latest advertised client limits
+- Nothing preventing a join is active (e.g. a hold-down timer,
+  administrative blocking, etc.)
 ~~~
 {: #fig-client-channel-states title="States a channel from the clients point of view."}
 
 {{fig-frame-exchange}} shows the frames that are being exchanged over the lifetime of an example channel.
 
 ~~~
-Client                                                                      Server
-
-                                                            <- MC_CHANNEL_ANNOUNCE: ID: 50, (...)
-                                                      ...
-                                                            <- MC_CHANNEL_PROPERTIES: ID: 50, (...)
-                                                            <- MC_CHANNEL_JOIN: ID: 50, (...)
-MC_CHANNEL_STATE: ID: 50, State: Joined, (...)    ->
-                                                            <- MC_CHANNEL_INTEGRITY: ID: 50, (...)
-                                                            <- [STREAM: (...)]
-MC_CHANNEL_ACK: ID: 50, (...)                     ->
-                                                      ...
-                                                           <- {[MC_CHANNEL_INTEGRITY: ID: 50, (...)]}
-                                                      ...
-                                                           <- {[MC_CHANNEL_PROPERTIES: ID: 50, (...)]}
-                                                      ...
-                                                           <- {[MC_CHANNEL_LEAVE: ID: 50, (...)]}
-{MC_CHANNEL_STATE: ID: 50, State: Left, (...)}    ->
-                                                      ...
-                                                           <- {MC_CHANNEL_JOIN: ID: 50, (...)}
-{MC_CHANNEL_STATE: ID: 50, State: Joined, (...)}  ->
-                                                      ...
-                                                           <- [MC_CHANNEL_RETIRE: ID: 50, (...)]
-MC_CHANNEL_STATE: ID: 50, State: Retired, (...)   ->
+Client                                        Server
+
+MC_LIMITS/initial_limits  --->
+
+                                              MC_ANNOUNCE
+                                              MC_KEY
+                                       <----  MC_JOIN
+
+MC_STATE(Joined)  --->
+
+                                              MC_INTEGRITY
+                                       <----  [STREAM(...)]
+MC_ACK  --->                                  ...
+...                                    <----  MC_KEY
+...
+MC_LIMITS  --->
+
+                                       <----  MC_LEAVE
+
+MC_STATE(Left)  --->
+
+                                       <----  MC_JOIN
+
+MC_STATE(Joined)  --->
+
+                                              MC_INTEGRITY
+                                       <----  [STREAM(...)]
+MC_ACK  --->                                  ...
+...
```

Comment:
I think the "..." got misaligned somehow?

Reply:
I have some on each side, I don't see them misaligned?
Maybe the picture is unclear tho.  It's meant to represent ongoing back & forth or something.  Not sure it conveys that well.

Reply:
Ah, I was following the examples in RFC9000 where I think they just had them in the middle and not on both sides, but I guess that works too